### PR TITLE
Fix/Optimize test_shouldBeWatered() which might break down test.

### PR DESCRIPTION
**Pull request by XinyueZ**
- Optimize shouldBeWatered()
- Changes of test_shouldBeWatered()

### CODE REVIEW

**Comment by XinyueZ** [comment]
@tiembo This PR should have fixed potential problem of `test_shouldBeWatered`.

File: app/src/test/java/com/google/samples/apps/sunflower/data/PlantTest.kt [hunk 1]
```diff
@@ -39,10 +40,37 @@ class PlantTest {
...
+            assertFalse(plant.shouldBeWatered(now, lastWateringDate.apply { add(DAY_OF_YEAR, -2) }))
+
+            // Test for lastWateringDate is some days ago, three days ago, four days ago etc.
+            lastWateringDate.time = now.time
+            val days = now.getMoreThanTwoDaysFromNow()
+            assertTrue(plant.shouldBeWatered(now, lastWateringDate.apply { add(DAY_OF_YEAR, -days) }))
+        }
+    }
+
+    private fun Calendar.getMoreThanTwoDaysFromNow(): Int {
```
Comment (tiembo) [inline app/src/test/java/com/google/samples/apps/sunflower/data/PlantTest.kt]: I think that using `Random()` here can cause unpredictable test failures should the details of `plant` change.  Prefer to replace `-days` in line 62 with `-3`.

**Comment by XinyueZ** [comment]
@tiembo check it

**Comment by tiembo** [comment]
Thank you @XinyueZ !

**Pull request merged**